Jint/Runtime/Interop/DelegateWrapper.cs

```diff
@@ -37,6 +38,12 @@ public DelegateWrapper(Engine engine, Delegate d)
         public override JsValue Call(JsValue thisObject, JsValue[] jsArguments)
         {
             var parameterInfos = _d.Method.GetParameters();
+
+            if (parameterInfos.Length > 0 && parameterInfos[0].ParameterType.FullName == "System.Runtime.CompilerServices.Closure")
+            {
+                parameterInfos = parameterInfos.Skip(1).ToArray();
```

Comment:
Skip is a bit nasty as it uses LINQ and doesn't perform that well, especially on full framework, but this is fortunately quite isolated case.

Reply:
I updated the code to more permanently use Array.Copy, it looks like it is about 0.1 ticks faster on average, but is a fair amount more complicated. Alternatively I could use ArraySegment<> (or Span<> in C# 7), but that would change the signature downstream and make the non-NETFRAMEWORK case less performant.

Reply:
Yes it was bit of nitpicking, in Jint codebase there should be as little `using System.Linq;` as possible, which I see you kept 😉 LINQ is much wiser on .NET Core and can be used more safely, but it still adds some penalty - especially on allocation side. So ideally following the Roslyn ideology of not using it at all if possible.

Thanks for considering this too.

Reply:
Fixed lol.

I'm used to code bases where an extra object allocation is the least of it's sins 😄